### PR TITLE
Feature/deprecation warnings

### PR DESCRIPTION
**Pull request by TkDodo**

### CODE REVIEW

File: packages/solid-query/src/QueryClientProvider.tsx [hunk 1]
```diff
@@ -80,7 +80,17 @@ export const QueryClientProvider = (
...
+    mergedProps.client.mount()
+
+    if (process.env.NODE_ENV !== 'production' && mergedProps.contextSharing) {
+      mergedProps.client
+        .getLogger()
+        .error(
+          `The contextSharing option has been deprecated and will be removed in the next major version`,
+        )
+    }
+  })
```
Comment (TkDodo) [inline packages/solid-query/src/QueryClientProvider.tsx:93]: @ardeora please let me know if this code is `solid` 😅

File: packages/vue-query/src/vueQueryPlugin.ts [hunk 2]
```diff
@@ -59,6 +59,14 @@ export const VueQueryPlugin = {
...
     client.mount()
 
+    if (process.env.NODE_ENV !== 'production' && options.contextSharing) {
+      client
+        .getLogger()
+        .error(
+          `The contextSharing option has been deprecated and will be removed in the next major version`,
+        )
+    }
+
```
Comment (TkDodo) [inline packages/vue-query/src/vueQueryPlugin.ts:69]: @DamianOsipiuk please let me know if this is the right place to do this for vue-query

Comment (TkDodo) [reply packages/vue-query/src/vueQueryPlugin.ts:69]: hmm for some reason, I'm getting an error that says `client.getLogger()` doesn't work. Do you know what I'm doing wrong?

```
packages/vue-query test:jest:   ● VueQueryPlugin › when context sharing is enabled › should use existing context
packages/vue-query test:jest:     TypeError: client.getLogger is not a function
packages/vue-query test:jest:       62 |     if (process.env.NODE_ENV !== 'production' && options.contextSharing) {
packages/vue-query test:jest:       63 |       client
packages/vue-query test:jest:     > 64 |         .getLogger()
packages/vue-query test:jest:          |          ^
packages/vue-query test:jest:       65 |         .error(
packages/vue-query test:jest:       66 |           `The contextSharing option has been deprecated and will be removed in the next major version`,
packages/vue-query test:jest:       67 |         )
```

Comment (DamianOsipiuk) [reply packages/vue-query/src/vueQueryPlugin.ts:69]: It's due to this mock, which has only one method and is not a real queryClient
https://github.com/TanStack/query/blob/1fe93566c43dd10d4f4adf307bfe337b6fc01a09/packages/vue-query/src/__tests__/vueQueryPlugin.test.ts#L248

Comment (TkDodo) [reply packages/vue-query/src/vueQueryPlugin.ts:69]: thx - fixed it here: 5f7ada6a

**Review by DamianOsipiuk — APPROVED** [review]

File: packages/solid-query/src/QueryClientProvider.tsx (hunk 1 above)
Comment (ardeora) [reply packages/solid-query/src/QueryClientProvider.tsx:93]: Hahahaha sorry for the late reply but yes! This is looking great. Shouldn't be an issue. 
lgtm

**Review by ardeora — APPROVED** [review]
Ran the tests locally. Nothing breaks and everything still looks super

**Pull request merged**